templates/admin.php

```diff
@@ -1,58 +1,23 @@
 <?php
 /** @var \OCP\IL10N $l */
 /** @var array $_ */
-use \OCP\App;
 
 script('passman', 'settings-admin');
 
 style('passman', 'admin');
 style('passman', 'vendor/font-awesome/font-awesome.min');
-
-$checkVersion = OC::$server->getConfig()->getAppValue('passman', 'check_version', '1') === '1';
-$AppInstance = new App();
-$localVersion = $AppInstance->getAppInfo("passman")["version"];
-$githubVersion = $l->t('Unable to get version info');
-if ($checkVersion) {
-	// get latest master version
-	$version = false;
-
-	$url = 'https://raw.githubusercontent.com/nextcloud/passman/master/appinfo/info.xml';
-	try {
-		$client = OC::$server->getHTTPClientService()->newClient();
-		$response = $client->get($url);
-		$xml = $response->getBody();
-	} catch (\Exception $e) {
-		$xml = false;
-	}
-
-	if ($xml) {
-		$loadEntities = libxml_disable_entity_loader(true);
-		$data = @simplexml_load_string($xml);
-		libxml_disable_entity_loader($loadEntities);
-		if ($data !== false) {
-			$version = (string)$data->version;
-		} else {
-			libxml_clear_errors();
-		}
-	}
-
-	if ($version !== false) {
-		$githubVersion = $version;
-	}
-}
-$ciphers = openssl_get_cipher_methods();
 ?>
 
 <div id="passwordSharingSettings" class="followup section">
 	<h2><?php p($l->t('Passman Settings')); ?></h2>
 	<?php
-	if ($checkVersion) {
-		p($l->t('GitHub version:') . ' ' . $githubVersion);
+	if ($_['checkVersion']) {
+		p($l->t('GitHub version:') . ' ' . $_['githubVersion']);
 		print '<br />';
 	} ?>
-	Local version: <?php p($localVersion); ?><br/>
+	Local version: <?php p($_['localVersion']); ?><br/>
 	<?php
-	if ($checkVersion && version_compare($githubVersion, $localVersion) === 1) {
+	if ($_['checkVersion'] && version_compare($_['githubVersion'], $_['localVersion']) === 1) {
 		p($l->t('A newer version of Passman is available'));
```

Comment:
Maybe we could show current and new version strings? (not a must)

Reply:
but if a new version is not available for the installed Nextcloud version, "new version" would also be confusing. So I would rather leave it that way.

Reply:
I mean, keep the if, but on the display text that says that there is a new version available would be cool to show the new version number, to make it translateable would be nice to have it with sprintf format, for example

Reply:
you mean the version also in the text like this?
![grafik](https://user-images.githubusercontent.com/30630233/112198824-cd31f980-8c0d-11eb-89b6-5e5a81ddeff1.png)

Reply:
ok, my brain completely ignored that the previous lines already contained that information 😅
Nothing to do around here! it perfect! :D